fix(linter): Fix `typescript/no-empty-interface` config option casing.

connorshea opened the pull request:
See the original rule's docs: https://typescript-eslint.io/rules/no-empty-interface/

This config option was incorrectly set up as using snake case, when it should've always been camelCase.

I added an alias so that `allow_single_extends` still works, in case anyone set it up like that in their linter config. But the docs have never mentioned it under that name.

camc314 commented:
### Merge activity

* **Nov 29, 9:08 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 29, 9:08 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Nov 29, 9:12 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
fix(linter): Fix `typescript/no-empty-interface` config option casing. (#16282)

See the original rule's docs: https://typescript-eslint.io/rules/no-empty-interface/

This config option was incorrectly set up as using snake case, when it should've always been camelCase.

I added an alias so that `allow_single_extends` still works, in case anyone set it up like that in their linter config. But the docs have never mentioned it under that name.
